core/server/server.js

```diff
@@ -477,6 +477,17 @@ class Server {
     this.registerServices()
 
     camp.timeout = this.config.public.requestTimeoutSeconds * 1000
+    if (this.config.public.requestTimeoutSeconds > 0) {
+      camp.on('timeout', socket => {
+        socket.write('HTTP/1.1 408 Request Timeout\n')
+        socket.write('Content-Type: text/html; charset=UTF-8\n')
+        socket.write('Content-Encoding: UTF-8\n')
+        socket.write('Cache-Control: max-age=60, s-maxage=60\n')
```

Comment:
I agonised a bit about what to do here.
On the one hand, it is kind of odd to cache this because repeat requests for any URLs that throw this will return a `Request Timeout` response.. immediately 🙃 That's kind of odd, but requests that tie us up for `requestTimeoutSeconds` seem like exactly the kind of request we don't want to immediately try and process at the origin again. I decided to tell downstream cache it for a bit.

Reply:
Makes sense, however, a minute does seem slightly high to me. Maybe we could make this configurable?

Reply:
On today's episode of the "naming things is hard" show, I've converted this to a snappily named setting `REQUEST_TIMEOUT_MAX_AGE_SECONDS` and switched the default to 30 seconds.